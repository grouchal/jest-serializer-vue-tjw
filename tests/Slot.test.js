const helpers = require('./helpers.js');

import { mount } from '@vue/test-utils';
import Slot from './components/Slot.vue';

describe('Slot.vue', () => {
  beforeEach(() => {
    helpers.mockSettings({});
  });

  test('Slot with default text', () => {
    const wrapper = mount(Slot, {
      slot: {
        default: 'test'
      }
    });

    expect(wrapper.html())
      .toEqual('<button>test</button>');

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Slot with markup', () => {
    const wrapper = mount(Slot, {
      slots: {
        default: '<p>test</p>'
      }
    });

    expect(wrapper.html())
      .toEqual('<button>\n  <p>test</p>\n</button>');

    expect(wrapper)
      .toMatchSnapshot();
  });
});
