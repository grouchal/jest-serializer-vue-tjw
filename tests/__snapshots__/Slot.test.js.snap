// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slot.vue Slot with default text 1`] = `<button>test</button>`;

exports[`Slot.vue Slot with markup 1`] = `
<button>
  <p>test</p>
</button>
`;
